src/main/java/org/broadinstitute/hellbender/utils/haplotype/Haplotype.java

```diff
@@ -98,11 +98,21 @@ public Haplotype trim(final Locatable loc) {
             return null;
         }
 
-        // note: trimCigarByReference does not remove leading indels, while getBasesCoveringRefInterval does remove bases
-        // of leading indels.  We must remove leading indels from the Cigar manually.
+        // note: trimCigarByReference does not remove leading or trailing indels, while getBasesCoveringRefInterval does remove bases
```

Comment:
I wrote a test for insertions.  The existing tests already cover the expected `null` result for deletions.

Reply:
We may at some point want to rethink the `null` result for leading/trailing deletions, however.  I get that if we are only trimming the padding after SW alignment of the padded haplotypes that every haplotype should share the initial (final) kmer of the ref source (sink).  In that a leading deletion means a bogus haplotype.  But I see nothing inherently wrong with a leading/trailing deletion of a haplotype that has been trimmed to fit a small genotyping window for Pair-HMM.  In practice since we pad these windows around all assembled events maybe no real variation would ever get lost in this way.  Still, it seems wrong and unnecessary.